src/datadog/grok/filters/keyvalue.rs

```diff
@@ -25,98 +26,152 @@ use super::super::{
     parse_grok_rules::Error as GrokStaticError,
 };
 
-static DEFAULT_FILTER_RE: Lazy<regex::Regex> = Lazy::new(|| Regex::new(r"^[\w.\-_@]*").unwrap());
+const DEFAULT_VALUE_RE: &str = r"\w.\-_@";
+const DEFAULT_DELIMITERS: (&str, &str) = (r"\s,;(\[{", r"\s,;)\]}");
+
+const DEFAULT_KEYVALUE_DELIMITER: &str = "=";
+const DEFAULT_QUOTES: &[(char, char)] = &[('"', '"'), ('\'', '\''), ('<', '>')];
 
 pub fn filter_from_function(f: &Function) -> Result<GrokFilter, GrokStaticError> {
     {
         let args_len = f.args.as_ref().map_or(0, |args| args.len());
 
         let key_value_delimiter = if args_len > 0 {
             match f.args.as_ref().unwrap()[0] {
-                FunctionArgument::Arg(Value::Bytes(ref bytes)) => {
-                    String::from_utf8_lossy(bytes).to_string()
-                }
+                FunctionArgument::Arg(Value::Bytes(ref bytes)) => &String::from_utf8_lossy(bytes),
                 _ => return Err(GrokStaticError::InvalidFunctionArguments(f.name.clone())),
             }
         } else {
-            // default key/value delimiter
-            "=".to_string()
+            DEFAULT_KEYVALUE_DELIMITER
         };
         let value_re = if args_len > 1 {
             match f.args.as_ref().unwrap()[1] {
                 FunctionArgument::Arg(Value::Bytes(ref bytes)) => {
-                    let mut re_str = String::new();
-                    re_str.push_str(r"^[\w.\-_@");
-                    re_str.push_str(&String::from_utf8_lossy(bytes));
-                    re_str.push_str("]+");
-                    Regex::new(re_str.as_str())
-                        .map_err(|_e| GrokStaticError::InvalidFunctionArguments(f.name.clone()))?
+                    [DEFAULT_VALUE_RE, &String::from_utf8_lossy(bytes)].concat()
                 }
                 _ => return Err(GrokStaticError::InvalidFunctionArguments(f.name.clone())),
             }
         } else {
             // default allowed unescaped symbols
-            DEFAULT_FILTER_RE.clone()
+            DEFAULT_VALUE_RE.to_string()
         };
 
         let quotes = if args_len > 2 {
             match f.args.as_ref().unwrap()[2] {
                 FunctionArgument::Arg(Value::Bytes(ref bytes)) => {
-                    let quotes = String::from_utf8_lossy(bytes);
-                    if quotes.len() == 2 {
-                        let mut chars = quotes.chars();
-                        vec![(
-                            chars.next().expect("open quote"),
-                            chars.next().expect("closing quote"),
-                        )]
-                    } else if quotes.is_empty() {
-                        // default
-                        vec![('"', '"'), ('\'', '\''), ('<', '>')]
-                    } else {
-                        return Err(GrokStaticError::InvalidFunctionArguments(f.name.clone()));
+                    let pair = String::from_utf8_lossy(bytes);
```

Comment:
Not necessarily a blocker for this PR but -- Would it make sense to move this and the other similar sections, like the one retrieving the value for `field_delimiters`, into their own functions since they're pretty complex?